ql/instruments/makevanillaswap.cpp

```diff
@@ -72,15 +73,23 @@ namespace QuantLib {
             // if the evaluation date is not a business day
             // then move to the next business day
             refDate = floatCalendar_.adjust(refDate);
-            Date spotDate = floatCalendar_.advance(refDate,
-                                                   settlementDays_*Days);
+            // use index valueDate interface wherever possible to estimate spot date.
+			// Unless we pass an explicit settlementDays_ which does not match the index-defined number of fixing days.
+			Date spotDate;
+            if ((settlementDays_ == Null<Natural>()) || (settlementDays_ == iborIndex_->fixingDays())) {
```

Comment:
I would keep only the `(settlementDays_ == Null<Natural>())` here.  If one passes a number of settlement days explicitly, I would use them even if they happen to be the same as those of the index.  What do you think?

Reply:
The reason I did it like this, is because the swap rate helpers seem to be explicitly setting `settlementDays_` to the index-specific number of fixing days as a fall-back. So unless we update the rate helper definitions we would never have cases of `settlementDays_ == Null<Natural>()` in curve construction.   

I personally felt this extra filter I put in was okay, because my intuition is that in the "normal" case the index-specific implementation of the settlement rules is what a user would expect. 

What do you think?

Reply:
See [here] for example. (https://github.com/lballabio/QuantLib/blob/88a0a65a576a792e6fe1342bcf32ce994fbcc1a6/ql/termstructures/yield/ratehelpers.cpp#L739)

Reply:
You're right.  I'll see if I can modify the rate helpers, otherwise we'll keep the condition as you wrote it.